src/plugins/flow.js

```diff
@@ -105,28 +105,44 @@ pp.flowParseDeclareFunction = function (node) {
   return this.finishNode(node, "DeclareFunction");
 };
 
-pp.flowParseDeclare = function (node) {
-  if (this.match(tt._class)) {
+let flowParseDeclareOrType = function(node, allowModule, allowType) {
```

Comment:
I'm not a huge fan of how I did this (especially since `node` is potentially unused), but otherwise I'd be repeating a bunch of stuff.